### ISSUE TITLE
DataStore HasMany scenario 3 - Post with Comments Lazy Load failing on "comment3PostIDId"

### ISSUE THREAD

**Issue by lawmicha** [aws-amplify/amplify-ios#939]
**Describe the bug**
When I retrieve a post with comments, the lazy loading of the comments is failing.

**To Reproduce**
Provision API with conflict resolution for DataStore with the following schema

(HasMany) A Post that can have many comments
```
type Post3 @model {
  id: ID!
  title: String!
  comments: [Comment3] @connection(keyName: "byPost3", fields: ["id"])
}

type Comment3 @model
  @key(name: "byPost3", fields: ["postID", "content"]) {
  id: ID!
  postID: ID!
  content: String!
}
```
See https://docs.amplify.aws/cli/graphql-transformer/connection for more details

2. Create a Post, a Comment with the Post
3. Query for the Post, and lazy load the comments

See this "no such column: root.comment3PostIDId (code: 1)"

**Expected behavior**
Lazy loading the comments should be successful

**Comment by lawmicha** [comment]
Use this test as accpetance criteria for fixing this https://github.com/aws-amplify/amplify-ios/pull/926#discussion_r536261252

**Comment by lawmicha** [comment]
when lazyLoading, this should be the fix
```swift
        var name = associatedField.name // always use the name first, it's the explicit key field
        if case let .belongsTo(_, targetName) = associatedField.association {
            name = targetName ?? modelName.camelCased() + associatedField.name.pascalCased() + "Id"
        }

```

**Comment by lawmicha** [comment]
this can be closed after https://github.com/aws-amplify/amplify-ios/pull/965 is released

**Comment by wooj2** [comment]
This was released as part of v1.5.3 -- closing
